cmake/modules/dependency_notifier.cmake

```diff
@@ -0,0 +1,101 @@
+## SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+## INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+## CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+## ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+## POSSIBILITY OF SUCH DAMAGE.
+
+if(NOT CMAKE_C_COMPILER)
+    message("-- No C compiler was found. Please install the gcc package for your distribution:
+    DEB: apt-get install gcc
+    RPM: yum install gcc") 
+endif()
+
+if(NOT CMAKE_CXX_COMPILER)
+    message("-- No C++ compiler was found. Please install the g++ package for your distribution:
+    DEB: apt-get install g++
+    RPM: yum install gcc-c++") 
+endif()
+
+if(NOT libjson-c_FOUND)
+    message("-- json-c not found. Please install json-c package for you respective distribution:
+    DEB: apt-get install libjson0-dev
+    RPM: yum install json-c
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable LIBJSON-C_ROOT in 
+   your cmake command as follows: cmake <path to clone dir> -DLIBJSON-C_ROOT=<path to install location>")
+endif()
+
+if(NOT libuuid_FOUND)
+    message("-- uuid not found. Please install uuid package for your respective distribution:
+    DEB: apt-get install uuid-dev
+    RPM: yum install libuuid
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable LIBJSON-C_ROOT in 
+   your cmake command as follows: cmake <path to clone dir> -DUUID_ROOT=<path to install location>")
+endif()
+
+find_package(PythonInterp 2.7 REQUIRED)
+if(NOT PYTHONINTERP_FOUND )
+    message("-- No suitable python interpreter found. Please install Python >= 2.7 to satisfy dependency for building tools.")
+endif()
+
+if(NOT DOXYGEN_FOUND)
+    message("-- Doxygen not found. Documentation will not be built. If you want documentation to be built, please install doxygen:
+    DEB: apt-get install doxygen
+    RPM: yum install doxygen
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable DOXYGEN_ROOT in 
+   your cmake command as follows: cmake <path to clone dir> -DDOXYGEN_ROOT=<path to install location>")
+endif()
+if(NOT SPHINX_FOUND)
+    message("-- Sphinx not found. HTML documentation will not be built. If you want HTML documentation to be built, 
+   please install python-sphinx:
+    DEB: apt-get install python-sphinx
+    RPM: Please follow the official documentation to install Sphinx: http://www.sphinx-doc.org/en/stable/install.html
+   If you have already installed this package in a nonstandard location 
+   please specify the location by defining the variable SPHINX_ROOT in 
+   your cmake command as follows: cmake <path to clone dir> -DSPHINX_ROOT=<path to install location>")
+endif()
+
+
+if((NOT libjson-c_FOUND) AND (NOT libuuid_FOUND))
```

Comment:
Can this be simplified? Maybe have to cmake vars, REQUIRED_DEPENDENCIES and OPTIONAL_DEPENDENCIES. Then, above, for example, when UUID is not found you can do:
```cmake
set(REQUIRED_DEPENDENCIES "libuuid ${REQUIRED_DEPENDENCIES}")
```
Then, instead of checking the different combinations, simple state:
```cmake
message(FATAL_ERROR,
  "The following dependencies are required and libopae-c will not be built without them. ${REQUIRED_DEPENDENCIES}")
```

Just a suggestion...